pkg/apis/garden/validation/validation.go

```diff
@@ -295,7 +302,34 @@ func validateKubernetesConstraints(kubernetes garden.KubernetesConstraints, fldP
 	return allErrs
 }
 
-func validateMachineTypeConstraints(machineTypes []garden.MachineType, fldPath *field.Path) field.ErrorList {
+func validateKubernetesSettings(kubernetes garden.KubernetesSettings, fldPath *field.Path) field.ErrorList {
```

Comment:
Should we also have validation for `.spec.kubernetes.versions` for duplicate versions?
Currently you can have:
```yaml
spec:
  kubernetes:
    versions:
    - expirationDate: "2020-04-05T08:00:00Z"
      version: 1.10.13
    - expirationDate: "2007-04-05T08:00:00Z"
      version: 1.10.13
```

Reply:
Yes, we can do it 👍 but let's do it outside of this PR. This validation feature wasn't there before, and it's not critical to continue because the `CloudProfile`s are controlled by operators. Can you create an issue?